st2common/st2common/constants/triggers.py

```diff
@@ -252,40 +252,64 @@
             "type": "string"
         },
         "year": {
-            "type": "integer"
+            "anyOf": [
+                {"type": "string"},
+                {"type": "integer"}
+            ],
         },
         "month": {
-            "type": "integer",
+            "anyOf": [
+                {"type": "string"},
+                {"type": "integer"}
+            ],
             "minimum": 1,
```

Comment:
Not sure, I would need to test it.

In any case, it doesn't really matter since we now perform stricter validation inside the trigger validation code path which is executed on rule creation.